shared-bindings/wifi/Radio.c

```diff
@@ -89,7 +89,7 @@ STATIC mp_obj_t wifi_radio_set_hostname(mp_obj_t self_in, mp_obj_t hostname_in)
 
     mp_arg_validate_length_range(hostname.len, 1, 253, MP_QSTR_hostname);
 
-    #ifndef CONFIG_IDF_TARGET_ESP32C3
+    #if 0 // ndef CONFIG_IDF_TARGET_ESP32C3
```

Comment:
Does this break ESP32-C3 support?

Reply:
ESP32-C3 and RP2040 don't have the required 'regex' library.

This change has the effect of disabling checking the hostname format on esp32, esp32-s2, esp32-s3. Better would be to wrtie a check that uses the "re" module's code.